### ISSUE TITLE
2.12 compiler no longer works on IBM JVM

### ISSUE THREAD

**Issue by shangd** [scala/bug#10417]
Cannot even start the Scala console
```
> scala
Welcome to Scala 2.12.2 (IBM J9 VM, Java 1.8.0).
Type in expressions for evaluation. Or try :help.

scala> #0: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x8cf18f) [0x7fffe963718f]
#1: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x1095d1) [0x7fffe8e715d1]
#2: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9prt28.so(+0x3ec7e) [0x7ffff573ec7e]
#3: /lib64/libpthread.so.0(+0xf370) [0x7ffff7bd0370]
#4: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2f4f79) [0x7fffe905cf79]
#5: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2f7306) [0x7fffe905f306]
#6: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2f779b) [0x7fffe905f79b]
#7: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2f98df) [0x7fffe90618df]
#8: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x69655f) [0x7fffe93fe55f]
#9: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2ee81d) [0x7fffe905681d]
#10: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x2efdb4) [0x7fffe9057db4]
#11: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x7eac2b) [0x7fffe9552c2b]
#12: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x7f25fc) [0x7fffe955a5fc]
#13: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x11158d) [0x7fffe8e7958d]
#14: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x112aa0) [0x7fffe8e7aaa0]
#15: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9prt28.so(+0x3f005) [0x7ffff573f005]
#16: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x114cc0) [0x7fffe8e7ccc0]
#17: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x115926) [0x7fffe8e7d926]
#18: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9prt28.so(+0x3f005) [0x7ffff573f005]
#19: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so(+0x1161f4) [0x7fffe8e7e1f4]
#20: /opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9thr28.so(+0x99cc) [0x7ffff5b7f9cc]
#21: /lib64/libpthread.so.0(+0x7dc5) [0x7ffff7bc8dc5]
#22: function clone+0x6d [0x7ffff74dc73d]
Unhandled exception
Type=Segmentation error vmState=0x000507ff
J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
Handler1=00007FFFF5E07140 Handler2=00007FFFF573EA80 InaccessibleAddress=0000000000000008
RDI=00007FFFE633D4A0 RSI=00007FFFE62B56B8 RAX=0000000000000000 RBX=00007FFFE62D0510
RCX=00007FFFE9AA3B00 RDX=0000000000000156 R8=00007FFFE62D8180 R9=00007FFFE96D037E
R10=000000000000FFE0 R11=0000000000000001 R12=00007FFFE7AA83F0 R13=00007FFFE633D4A0
R14=00007FFFE62BFBB8 R15=00007FFFE96D037E
RIP=00007FFFE905CF79 GS=0000 FS=0000 RSP=00007FFFE7AA82B0
EFlags=0000000000210246 CS=0033 RBP=00007FFFE62D89A0 ERR=0000000000000004
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=0000000000000008
xmm0 000a000900070006 (f: 458758.000000, d: 1.390690e-308)
xmm1 0009000700060005 (f: 393221.000000, d: 1.251619e-308)
xmm2 00007fffe62e91a0 (f: 3861811712.000000, d: 6.953334e-310)
xmm3 00007fffe62e9220 (f: 3861811712.000000, d: 6.953334e-310)
xmm4 00007fffe62e92a0 (f: 3861811968.000000, d: 6.953334e-310)
xmm5 00007fffe6362498 (f: 3862308096.000000, d: 6.953334e-310)
xmm6 2020202020202020 (f: 538976256.000000, d: 6.013470e-154)
xmm7 00ffffffffffffff (f: 4294967296.000000, d: 7.291122e-304)
xmm8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm9 00ffffffffffff00 (f: 4294967040.000000, d: 7.291122e-304)
xmm10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm15 402e7f9c1e980d00 (f: 513281280.000000, d: 1.524924e+01)
Module=/opt/ibm/java-x86_64-80/jre/lib/amd64/compressedrefs/libj9jit28.so
Module_base_address=00007FFFE8D68000

Method_being_compiled=scala/collection/mutable/HashMap.elemHashCode(Ljava/lang/Object;)I
Target=2_80_20170616_352529 (Linux 3.10.0-514.16.1.el7.x86_64)
CPU=amd64 (2 logical CPUs) (0xecf43000 RAM)
----------- Stack Backtrace -----------
(0x00007FFFE905CF79 [libj9jit28.so+0x2f4f79])
(0x00007FFFE905F306 [libj9jit28.so+0x2f7306])
(0x00007FFFE905F79B [libj9jit28.so+0x2f779b])
(0x00007FFFE90618DF [libj9jit28.so+0x2f98df])
(0x00007FFFE93FE55F [libj9jit28.so+0x69655f])
(0x00007FFFE905681D [libj9jit28.so+0x2ee81d])
(0x00007FFFE9057DB4 [libj9jit28.so+0x2efdb4])
(0x00007FFFE9552C2B [libj9jit28.so+0x7eac2b])
(0x00007FFFE955A5FC [libj9jit28.so+0x7f25fc])
(0x00007FFFE8E7958D [libj9jit28.so+0x11158d])
(0x00007FFFE8E7AAA0 [libj9jit28.so+0x112aa0])
(0x00007FFFF573F005 [libj9prt28.so+0x3f005])
(0x00007FFFE8E7CCC0 [libj9jit28.so+0x114cc0])
(0x00007FFFE8E7D926 [libj9jit28.so+0x115926])
(0x00007FFFF573F005 [libj9prt28.so+0x3f005])
(0x00007FFFE8E7E1F4 [libj9jit28.so+0x1161f4])
(0x00007FFFF5B7F9CC [libj9thr28.so+0x99cc])
(0x00007FFFF7BC8DC5 [libpthread.so.0+0x7dc5])
clone+0x6d (0x00007FFFF74DC73D [libc.so.6+0xf773d])
---------------------------------------
JVMDUMP039I Processing dump event "gpf", detail "" at 2017/07/19 01:40:56 - please wait.
JVMDUMP032I JVM requested System dump using '//core.20170719.014056.21454.0001.dmp' in response to an event
JVMDUMP010I System dump written to //core.20170719.014056.21454.0001.dmp
JVMDUMP032I JVM requested Java dump using '//javacore.20170719.014056.21454.0002.txt' in response to an event
JVMDUMP010I Java dump written to //javacore.20170719.014056.21454.0002.txt
JVMDUMP032I JVM requested Snap dump using '//Snap.20170719.014056.21454.0003.trc' in response to an event
JVMDUMP010I Snap dump written to //Snap.20170719.014056.21454.0003.trc
JVMDUMP007I JVM Requesting JIT dump using '//jitdump.20170719.014056.21454.0004.dmp'
JVMDUMP010I JIT dump written to //jitdump.20170719.014056.21454.0004.dmp
JVMDUMP013I Processed dump event "gpf", detail "".
```

JVM version:
IBM Java SDK8 8.0.4.7 – 30 June 2017
```
> java -version
java version "1.8.0"
Java(TM) SE Runtime Environment (build pxa6480sr4fp7-20170627_02(SR4 FP7))
IBM J9 VM (build 2.8, JRE 1.8.0 Linux amd64-64 Compressed References 20170616_352529 (JIT enabled, AOT enabled)
J9VM - R28_20170616_0201_B352529
JIT  - tr.r14.java_20170616_352529
GC   - R28_20170616_0201_B352529_CMPRSS
J9CL - 20170616_352529)
JCL - 20170622_01 based on Oracle jdk8u131-b11
```
https://developer.ibm.com/javasdk/downloads/sdk8/

It works on the previous Scala version 2.11.11, but not on 2.12.2 nor 2.13.0-M1. 

It also works on other JVM (openjdk/oracle jdk), not sure if this is a bug in the IBM JVM.

**Comment by lrytz** [comment]
That certainly looks like a bug in J9, any idea where to report it?

**Comment by shangd** [comment]
The only place I can find is the forum, so I opened a topic there as well
https://www.ibm.com/developerworks/community/forums/html/topic?id=fd19d251-50bd-4c19-b193-4b167c7990ac&ps=25

**Comment by eli-jordan** [comment]
@SethTisue This is not related to jdk9, but rather to j9 the IBM Implementation of the jvm, running jdk8

**Comment by bjornvar** [comment]
Thank you very much for bringing this to our attention! We're investigating the issue.
(From the J9 JVM development team at IBM)

**Comment by bjornvar** [comment]
The issue has been resolved, and the fix will be available in the next release (8.0.5.0).

**Comment by lrytz** [comment]
Thank you!
